commit a3a82a148df4dddcb81748a06e7e8b70c43b3610
Author: Krister Johansen <kjlx@templeofstupid.com>
Date:   Wed Jan 4 01:22:52 2017 -0800

    dmaengine: iota: ioat_alloc_chan_resources should not perform sleeping allocations.
    
    commit 21d25f6a4217e755906cb548b55ddab39d0e88b9 upstream.
    
    On a kernel with DEBUG_LOCKS, ioat_free_chan_resources triggers an
    in_interrupt() warning.  With PROVE_LOCKING, it reports detecting a
    SOFTIRQ-safe to SOFTIRQ-unsafe lock ordering in the same code path.
    
    This is because dma_generic_alloc_coherent() checks if the GFP flags
    permit blocking.  It allocates from different subsystems if blocking is
    permitted.  The free path knows how to return the memory to the correct
    allocator.  If GFP_KERNEL is specified then the alloc and free end up
    going through cma_alloc(), which uses mutexes.
    
    Given that ioat_free_chan_resources() can be called in interrupt
    context, ioat_alloc_chan_resources() must specify GFP_NOWAIT so that the
    allocations do not block and instead use an allocator that uses
    spinlocks.
    
    Signed-off-by: Krister Johansen <kjlx@templeofstupid.com>
    Acked-by: Dave Jiang <dave.jiang@intel.com>
    Signed-off-by: Vinod Koul <vinod.koul@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1412_a3a82a148df4dddcb81748a06e7e8b70c43b3610.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1412_a3a82a148df4dddcb81748a06e7e8b70c43b3610.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
